.github/workflows/test.yml

```diff
@@ -0,0 +1,60 @@
+on:
+  pull_request:
+    branches:
+      - main
+      - master
+  push:
+    branches:
+      - main
+      - master
+  release:
+    types: [published]
+
+env:
+  GOFLAGS: "-mod=vendor"
+  GO111MODULE: on
+
+jobs:
+  test:
+    name: Test and Publish
+    runs-on: ubuntu-20.04
+    strategy:
+      fail-fast: false
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v4
+
+      - uses: actions/setup-go@v5
+        with:
+          go-version: '1.16'
+
+      - name: Setup
+        run: |
+          make tools
+          go mod vendor
+
+      - name: Code Lint
+        run:  make lint
+
+      - name: Code UnitTest
+        run: make test
+        if: ${{ github.event_name == 'release' }}
```

Comment:
Change requested: These aren't the terraform acceptance tests that we only want to run during integration. Unit tests are normal go tests that we want to run on pushes and PRs as well.

```suggestion
```

Reply:
Bad copy paste, good eye.